src/main/java/bio/terra/service/tabulardata/google/bigquery/BigQueryPdao.java

```diff
@@ -252,4 +251,154 @@ public static List<BigQueryDataResultModel> aggregateTableData(TableResult resul
             });
     return values;
   }
+
+  // COLUMN STATS
+  public static final String ARRAY_TEXT_COLUMN_STATS_TEMPLATE =
+      """
+          WITH array_field AS (Select <column> FROM <table> <whereClause>)
+            Select flattened_array_field AS <column>, COUNT(*) AS <countColumn> FROM array_field CROSS JOIN UNNEST(array_field.<column>)
+            AS flattened_array_field GROUP BY flattened_array_field ORDER BY flattened_array_field <direction>
+          """;
+  public static final String TEXT_COLUMN_STATS_TEMPLATE =
+      """
+        SELECT <column>, COUNT(*) AS <countColumn> FROM <table> AS <tableName> <whereClause> GROUP BY <tableName>.<column> ORDER BY <column> <direction>
```

Comment:
Do we want to order by column name or by count (in descending order).  That'll allow some easier filtering in the UI where we only show say, the top 10 values

Reply:
I think ordering by column value in alphabetic order makes it easier for the user to find elements in UI. In the UI, we always provide the user with all of the elements. (We just switch how they're presented if they hit the 30 element threshold). Looking at the existing UI code, we don't order by anything, so there isn't precedence here. I'm inclined to leave this as is (where it's ordered by column value)